pytest_splunk_addon/standard_lib/index_tests/test_templates.py

```diff
@@ -58,36 +50,89 @@ def test_index_time_extractions(
 
         if not results:
             assert False, "No Events found for query "+search
-
+        result_fields = dict()
         for result in results:
-            for key in result:
+            for key, val in result.items():
+                try:
+                    result_fields[key].append(val)
+                except KeyError:
+                    result_fields[key] = [val]
+
+        # This logic helps in comparing Results when the token is
+        # only replaced once but the value is assigned to n events
+        # Example syslog: all the headers are only tokenized once hence
+        #   key_fields = {'host': ['dummy_host']}
+        #   result_dict = {'host': ['dummy_host']*n}
+
+        assert (
+            {i: list(
+                set(result_fields)) for i in result_fields} == {i: list(
+                    set(fields_to_check)) for i in fields_to_check}
+        )
```

Comment:
Need to add failure message.

Reply:
On dictionary mismatch assertion error shows what values were mismatched, do we need specific message here?